src/guide/scaling-up/testing.md

```diff
@@ -49,30 +92,75 @@ Vitest is relatively new and is still undergoing rapid development. While it is
 
 ## Component Testing
 
-In Vue applications, components are the main building blocks of the UI. Components are therefore the natural unit of isolation when it comes to visual and interaction tests. From a granularity perspective, component testing sits somewhere above unit testing and can be considered a form of integration testing. In some cases we will be testing a single component, but in other cases we could be testing a tree of components to make sure they are integrated correctly.
+In Vue applications, components are the main building blocks of the UI. Components are therefore the natural unit of isolation when it comes to validating your application's behavior. From a granularity perspective, component testing sits somewhere above unit testing and can be considered a form of integration testing. Much of your Vue Application should be covered by a component test and we recommend that each Vue component has its own spec file.
+
+Component tests should catch issues relating to your component's props, events, slots that it provides, styles, classes, lifecycle hooks, and more.
 
-Component tests should focus on the component's public interfaces rather than internal implementation details. Or, in other words, **test what a component does, not how it does it**.
+They should not mock child components, but instead test the interactions between your component and its children by interacting with the components as a user would. For example, a component test should click on an element like a user would instead of programmatically interacting with the component.
 
-- **DO**
+Component tests should focus on the component's public interfaces rather than internal implementation details. For most components, the public interface is limited to: events emitted, props, and slots. When testing, remember to **test what a component does, not how it does it**.
 
-  - For **Visual** tests: assert correct render output based on input props and slots.
-  - For **Interaction** tests: assert correct render updates or emitted events in response to user input events.
+**DO**
+
+- For **Visual** logic: assert correct render output based on inputted props and slots.
+- For **Behavioral** logic: assert correct render updates or emitted events in response to user input events.
+
+In the below example, we demonstrate a Stepper component that has a DOM element labeled "increment" and can be clicked. We pass a prop called `max` that prevents the Stepper from being incremented past `2`, so if we click the button 3 times, the UI should still say `2`.
+
+We know nothing about the implementation of Stepper, only that the "input" is the `max` prop and the "output" is the state of the DOM as the user will see it.
+
+```js
+const valueSelector = '[data-testid=stepper-value]'
```

Comment:
I think we should include "this example is written using Cypress" before we present it, at the very least, since the style is really different from the unit test example (eg the assertions) and that might be a little confusing/jarring.

Reply:
100000% if we keep it as a Cy example, we should annotate it. It's disjointed and jarring like "Wtf is this syntax".

Reply:
I would go even further and suggest rewriting the test to Jest/having Jest code as an alternative here

Reply:
Any reason why this is written in Cy? Considering `vitest` as a first option why not start with "just pure vitest" - this will require people not to learn any additional syntax (Cy as an example) just to understand the test

Reply:
Jest and Vitest are the test runners and aren't really "visible" in this code example.

Vitest supports both Mocha and Jasmine matchers.

The debate is really:

<details>
<summary>What component mounting strategy are we recommending?</summary>

### Vue Test Utils

```js
const wrapper = mount(Stepper, {
  props: {
    max: 10
  }
})

```

### Cypress

```js
mount(Stepper, {
  props: {
    max: 10
  }
})
```

### Testing Library

```js
render(Stepper, {
  props: {
    max: 10
  }
})
```

</details>

<details>
<summary>What will we recommend for interacting with components? (Vue Test Utils, Testing Library, Cypress)</summary>

### Vue Test Utils

```js
await wrapper.get('[data-testid=increment]').trigger('click')
expect(wrapper.get('[data-testid=stepper]').text()).toEqual('1')
```

### Cypress

```js
cy.get('[data-testid=increment]').click()
cy.get('[data-testid=stepper]').should('contain.text', '1')
```

### Testing Library

```js
const button = screen.getByTestId('increment')
await fireEvent.click(button)
expect(screen.queryByText('1')).toBeTruthy()
```


</details>


<details>
<summary>What assertion library are we recommending? (Jasmine, Mocha, Cypress's wrapped Mocha)</summary>

### Jasmine

```js
expect('whatever').toBeSomething()
```

### Chai

```js
expect('whatever').to.be.something
```

### Cypress's Chai

```js
cy.get('button').should('be.something')
```


</details>

In this PR we recommend Vitest as the test runner for headless components and unit tests, which comes with both Mocha and Jasmine assertions (Jasmine is camel-case, Mocha is dots) and we recommend Cypress for mounting and clicking on headed components.

If we don't want users using `@vue/test-utils` directly, what syntax should we pick?

Reply:
I added _all the things_ in a tabbed component 😼. This should be helpful.

<img width="640" alt="Screen Shot 2022-02-07 at 9 39 20 AM" src="https://user-images.githubusercontent.com/2801156/152809714-4b9a6975-5381-4ef9-a18f-143433f4bd06.png">